custom-templates/priv_instances_public_nlb_ssl/neo4j-nlb-priv-ssl.template.yaml

```diff
@@ -0,0 +1,857 @@
+        - Key: Name
+          Value: Neo4jPrivateRouteTable
+      VpcId: !Ref Neo4jVPC
+
+  #For Private Subnets
+  Neo4jPrivateRoute:
+    Type: AWS::EC2::Route
+    Properties:
+      NatGatewayId: !Ref Neo4jNATGateway
+      RouteTableId: !Ref Neo4jPrivateRouteTable
+      DestinationCidrBlock: 0.0.0.0/0
+
+  #For Public Subnet
+  Neo4jPublicRoute:
+    Type: AWS::EC2::Route
+    Properties:
+      GatewayId: !Ref Neo4jIGW
+      RouteTableId: !Ref Neo4jPublicRouteTable
+      DestinationCidrBlock: 0.0.0.0/0
+
+  #For Public Subnet 1
+  Neo4jPublicSubnetRouteTableAssociation1:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref Neo4jPublicRouteTable
+      SubnetId: !Ref Neo4jPublicSubnet1
+
+  #For Public Subnet 2
+  Neo4jPublicSubnetRouteTableAssociation2:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref Neo4jPublicRouteTable
+      SubnetId: !Ref Neo4jPublicSubnet2
+
+  #For Public Subnet 3
+  Neo4jPublicSubnetRouteTableAssociation3:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref Neo4jPublicRouteTable
+      SubnetId: !Ref Neo4jPublicSubnet3
+
+  #For Private Subnet
+  Neo4jPrivateSubnetRouteTableAssociation1:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref Neo4jPrivateRouteTable
+      SubnetId: !Ref Neo4jPrivateSubnet1
+
+  #For Private Subnets
+  Neo4jPrivateSubnetRouteTableAssociation2:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref Neo4jPrivateRouteTable
+      SubnetId: !Ref Neo4jPrivateSubnet2
+
+  #For Private Subnets
+  Neo4jPrivateSubnetRouteTableAssociation3:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref Neo4jPrivateRouteTable
+      SubnetId: !Ref Neo4jPrivateSubnet3
+
+  #For Private Subnets
+  Neo4jNATGateway:
+    Type: AWS::EC2::NatGateway
+    Properties:
+      AllocationId: !GetAtt Neo4jNATGatewayEIP.AllocationId
+      SubnetId: !Ref Neo4jPublicSubnet1
+      Tags:
+        - Key: StackID
+          Value: !Ref 'AWS::StackId'
+        - Key: Name
+          Value: !Ref 'AWS::StackName'
+
+ #For Private Subnets
+  Neo4jNATGatewayEIP:
+    Type: AWS::EC2::EIP
+    Properties:
+      Domain: vpc
+      Tags:
+        - Key: StackID
+          Value: !Ref 'AWS::StackId'
+        - Key: Name
+          Value: !Ref 'AWS::StackName'
+
+  #VPC/Instance Security Group
+  Neo4jInstanceSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Security group for instances, open across the VPC
+      VpcId: !Ref Neo4jVPC
+      SecurityGroupIngress:
+        - IpProtocol: -1
+          CidrIp: !GetAtt Neo4jVPC.CidrBlock
+        - IpProtocol: tcp
+          FromPort: 7474
+          ToPort: 7474
+          CidrIp: 0.0.0.0/0
+        - IpProtocol: tcp
+          FromPort: 7687
+          ToPort: 7687
+          CidrIp: 0.0.0.0/0
```

Comment:
Could this be configured to the CIDRs on the public subnets rather than 0.0.0.0/0?